javaslang/src/main/java/javaslang/collection/TreeMap.java

```diff
@@ -477,6 +505,98 @@ public boolean isEmpty() {
         return TreeSet.ofAll(keyComparator(), iterator().map(Tuple2::_1));
     }
 
+    @Override
+    public Iterator<Tuple2<K, V>> descendingIterator() {
+        return entries.descendingIterator();
+    }
+
+    @Override
+    public SortedSet<K> descendingKeySet() {
```

Comment:
* Return `TreeMap` instead of `SortedSet`
I will do. Returning `TreeMap` makes the interface nicer to use (you were working with a `TreeMap`, and after the operation, you continue working with another `TreeMap`)

* Sort methods alphabetically
Sorry, I wasn't aware that was the convention. I personally prefer sorting methods by responsibility, i.e. `headMap` and `tailMap` are closely related and thus should be together.  But it is far more important to be consistent, so if everything else is sorted alphabetically, we should continue doing so.